frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/ModelManager.java

```diff
@@ -41,21 +42,25 @@
     private ConfigManager configManager;
     private WorkLoadManager wlm;
     private ConcurrentHashMap<String, Model> models;
+    private Set<String> startupModels;
     private ScheduledExecutorService scheduler;
 
-    private ModelManager(ConfigManager configManager, WorkLoadManager wlm) {
+    private ModelManager(
+            ConfigManager configManager, WorkLoadManager wlm, Set<String> startupModels) {
```

Comment:
You don't need to pass startModels in constructor. You can create it in the constructor, it will be much cleaner.

In the initialization code, you simplely call: modelManager.getStartupModel().add(modelName).

Reply:
This seems cleaner as we don't need any API. This can be set up only during initialization of ModelManager and no where else.

Reply:
You already have getStartupModels() in ModelManager.

Reply:
Oh, that way. I think we should change this getter then. We shouldn't have a mean to write to this object post startup. I will make changes.

Reply:
I will fall back to mutable sets. Trying to make them immutable to "add" seems more complicated.